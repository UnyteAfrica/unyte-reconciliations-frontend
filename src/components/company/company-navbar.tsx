import { useContext, useEffect, useState } from "react";
import { NavLink, useLocation } from "react-router-dom";
import { BiChevronDown } from "react-icons/bi";
import { Icon } from "../shared/icon";
import { BrowserComboRoutes, BrowserRoutes } from "@/utils/routes";
import { cx } from "class-variance-authority";
import { twMerge } from "tailwind-merge";
import { clearCredentials, getCompanyInitials } from "@/utils/utils";
import { useQuery, useQueryClient } from "@tanstack/react-query";
import { CompanyQueryKeys } from "@/utils/query-keys";
import { Loader } from "../loader";
import { useLockScroll, useMediaQuery } from "@/utils/hooks";
import { LuMenu } from "react-icons/lu";
import { AiOutlineClose } from "react-icons/ai";
import { getProfile } from "@/services/api/api-base";
import { AuthContext } from "@/context/auth.context";

type UrlLink = {
  text: string;
  url: string;
};

const navLinks: UrlLink[] = [
  {
    text: "Overview",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.overview,
  },
  {
    text: "Policies",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.policies,
  },
  {
    text: "Claims",
    url: BrowserComboRoutes.pendingCompanyClaims,
  },
  {
    text: "Agents",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.agents,
  },
];

export const CompanyNavbar = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
  const location = useLocation();
  const { setIsLoggedIn, setEmail } = useContext(AuthContext);
  const queryClient = useQueryClient();

  const { data: companyDetailsData, isPending: isCompanyDetailsLoading } =
    useQuery({
      queryKey: [CompanyQueryKeys.profile],
      queryFn: () => getProfile(),
    });

  useEffect(() => {
    setIsMobileMenuOpen(false);
    setIsOpen(false);
  }, [location.pathname]);

  const logout = () => {
    clearCredentials();
    setIsLoggedIn(false);
    queryClient.invalidateQueries();
  };

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  const companyDetails = companyDetailsData?.data;

  useEffect(() => {
    if (companyDetails) setEmail(companyDetails.email);
  }, [companyDetails]);

  const { isMediaQueryMatched } = useMediaQuery(1024);

  useLockScroll(isMobileMenuOpen);

  return (
    <>
      {!isMediaQueryMatched && (
        <div className="relative z-30">
          <header className="px-5 pt-10 pb-4">
            <LuMenu
              onClick={() => setIsMobileMenuOpen(true)}
              size={32}
              className="block"
            />
            <Icon type="logo" className="block" />
          </header>
          <div
            className={twMerge(
              "absolute top-0 left-0 flex w-full transition duration-300 -translate-x-[100%]",
              isMobileMenuOpen && "translate-x-0"
            )}
          >
            <nav className="h-dvh bg-white w-[300px] relative px-5 py-10 flex flex-col justify-between">
              <div className="flex justify-between items-center">
                <Icon type="logo" className="block" />
                <AiOutlineClose
                  onClick={() => setIsMobileMenuOpen(false)}
                  className="right-5 w-6 h-6"
                />
              </div>
              <div className="flex flex-col space-y-8">
                {navLinks.map((navLink, idx) => (
                  <NavLink
                    key={idx}
                    to={navLink.url}
                    className={({ isActive }) =>
                      cx(
                        "font-semibold text-lg duration-300",
                        (isActive ||
                          location.pathname.includes(
                            navLink.text.toLowerCase()
                          )) &&
                          "text-[#25D366] underline underline-offset-[10px] decoration-4"
                      )
                    }
                  >
                    {navLink.text}
                  </NavLink>
                ))}
              </div>
              <div>
                {isCompanyDetailsLoading ? (
                  <Loader />
                ) : (
                  !!companyDetails && (
                    <div id="profile">
                      <div className="flex items-center mb-6">
                        {!!companyDetails.profile_image ? (
                          <img
                            className="h-10 w-10 object-cover inline-block mr-2 rounded-full"
                            src={companyDetails.profile_image}
                            alt=""
                          />
                        ) : (
                          <div className="rounded-full h-10 w-10 bg-gray-200 text-base flex items-center justify-center mr-2">
                            {getCompanyInitials(companyDetails.business_name)}
                          </div>
                        )}

                        <div className="text-base text-[#333] font-medium">
                          {companyDetails.business_name}
                        </div>
                      </div>

                      <button
                        className="block text-[#EB5757] font-medium"
                        onClick={logout}
                      >
                        Sign out
                      </button>
                    </div>
                  )
                )}
              </div>
            </nav>
            <div
              className="grow bg-black/20 h-dvh"
              onClick={() => setIsMobileMenuOpen(false)}
            />
          </div>
        </div>
      )}
      {isMediaQueryMatched && (
        <div
          id="navbar-container"
          className="flex justify-between items-center p-6 px-0 mx-auto max-w-6xl"
        >
          <div>
            <Icon className="shrink-0" type="logo" alt="unyte logo" />
          </div>
          <div className="space-x-24">
            {navLinks.map((navLink, idx) => {
              return (
                <NavLink
                  key={idx}
                  to={navLink.url}
                  className={({ isActive }) =>
                    cx(
                      "font-semibold text-lg duration-300",
                      (isActive ||
                        location.pathname.includes(
                          navLink.text.toLowerCase()
                        )) &&
                        "text-[#25D366] underline underline-offset-[30px] decoration-4"
                    )
                  }
                >
                  {navLink.text}
                </NavLink>
              );
            })}
          </div>

          {isCompanyDetailsLoading ? (
            <Loader />
          ) : (
            !!companyDetails && (
              <div id="profile">
                <div className="relative">
                  <button
                    className="px-4 py-2 text-[#333333] rounded-md"
                    onClick={toggleDropdown}
                  >
                    <div className="space-x-2 flex flex-row items-center">
                      {!!companyDetails.profile_image ? (
                        <img
                          className="rounded-full h-10 w-10 object-cover"
                          src={companyDetails.profile_image}
                          alt=""
                        />
                      ) : (
                        <div className="rounded-full h-10 w-10 p-2 bg-gray-200 text-base flex items-center justify-center">
                          {getCompanyInitials(companyDetails.business_name)}
                        </div>
                      )}
                      <span className="text-lg font-semibold hidden min-[1200px]:inline">
                        {companyDetails.business_name}
                      </span>{" "}
                      <BiChevronDown
                        className={cx(
                          "transition-all duration-300 w-6 h-6",
                          isOpen && "rotate-180"
                        )}
                      />
                    </div>
                  </button>
                  <div
                    className={twMerge(
                      "absolute w-[200px] top-[65px] right-0 mt-2 py-2 z-10 border border-[#e1e1e1] bg-white shadow-lg rounded-lg max-h-0 transition-all duration-300 overflow-hidden opacity-0",
                      isOpen && "max-h-[240px] opacity-100"
                    )}
                  >
                    <div className="px-6 py-4 flex items-center p-2">
                      <span className="text-base text-[#333] font-medium">
                        {companyDetails.business_name}
                      </span>
                    </div>
                    <hr />
                    <NavLink
                      to={
                        BrowserComboRoutes.companyDashboard +
                        BrowserRoutes.profile
                      }
                      className="block px-6 py-4 text-[#333] font-medium"
                    >
                      Profile
                    </NavLink>
                    <hr />
                    <button
                      onClick={logout}
                      className="block px-6 py-4 text-[#EB5757] font-medium"
                    >
                      Sign out
                    </button>
                  </div>
                </div>
              </div>
            )
          )}
        </div>
      )}

      <hr />
    </>
  );
};
