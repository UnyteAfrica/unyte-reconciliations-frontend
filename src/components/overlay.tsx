import { PropsWithChildren, useContext } from "react";
import { twMerge } from "tailwind-merge";
import { useLockScroll } from "@/utils/hooks";
import { OverlayContext, OverlayContextType } from "@/context/overlay.context";

export const Overlay: React.FC<PropsWithChildren<{ className?: string }>> = ({
  className,
  children,
}) => {
  const { isNewAgentOverlayOpened, setNewAgentOverlayOpened } = useContext(
    OverlayContext
  ) as OverlayContextType;

  useLockScroll(true);

  const handleClose = () => {
    if (isNewAgentOverlayOpened) setNewAgentOverlayOpened(false);
  };

  return (
    <div
      className={twMerge(
        "bg-[#080202]/[.40] h-screen w-screen flex justify-center items-center fixed top-0 left-0 z-40",
        className
      )}
      onClick={handleClose}
    >
      <div className="max-h-[80vh] overlayContainer w-[90%] max-w-[550px] overflow-y-auto rounded-2xl bg-white">
        {children}
      </div>
    </div>
  );
};
