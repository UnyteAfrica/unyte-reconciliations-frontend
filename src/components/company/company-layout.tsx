import { Outlet, useNavigate } from "react-router-dom";
import { CompanyNavbar } from "./company-navbar";
import { useContext, useEffect } from "react";
import { CompanyContext } from "@/context/company.context";
import { BrowserComboRoutes } from "@/utils/routes";
import { LocalStorage } from "@/services/local-storage";

export const CompanyLayout = () => {
  const { isLoggedIn, setIsLoggedIn } = useContext(CompanyContext);

  const navigate = useNavigate();

  const listener = (e: CustomEvent) => {
    if (e.detail.key == "companyAccessToken" && e.detail.value == null) {
      navigate(BrowserComboRoutes.companyLogin);
      setIsLoggedIn(false);
    }
  };

  useEffect(() => {
    window.addEventListener(LocalStorage.eventName, listener as EventListener);

    return () =>
      window.removeEventListener(
        LocalStorage.eventName,
        listener as EventListener
      );
  }, []);

  useEffect(() => {
    if (!isLoggedIn) navigate(BrowserComboRoutes.companyLogin);
  }, [isLoggedIn]);

  return (
    <div>
      <CompanyNavbar />
      <Outlet />
    </div>
  );
};
