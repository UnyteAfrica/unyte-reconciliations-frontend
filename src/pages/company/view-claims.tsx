import { nairaSign } from "@/utils/utils";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import car from "@/assets/images/car.png";
import { useContext } from "react";
import { OverlayContext, OverlayContextType } from "@/context/overlay.context";
import { Overlay } from "@/components/overlay";
import { ClaimStatusOverlay } from "@/components/overlays/claim-status-overlay";

const settings = {
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 1,
  slidesToScroll: 1,
};

export const ViewClaimsPage = () => {
  const { isUpdateClaimStatusOpened, setIsUpdateClaimStatusOpened } =
    useContext(OverlayContext) as OverlayContextType;

  return (
    <>
      {isUpdateClaimStatusOpened && (
        <Overlay>
          <ClaimStatusOverlay />
        </Overlay>
      )}
      <div className="p-8 max-w-6xl mx-auto w-full">
        <header className="flex pb-4 border-b border-[#e0e0e0] flex-col lg:flex-row lg:w-full lg:justify-between">
          <div className="flex flex-col space-y-3 mb-4">
            <div className="flex items-center">
              <em className="font-medium text-xl inline-block mr-4 not-italic">
                123jkf5402
              </em>
              <div className="p-2 py-1 font-medium text-sm w-20 text-center bg-[#4f4f4f]/[.1]">
                Submitted
              </div>
            </div>
            <em className="not-italic font-medium text-sm">May 7, 2023</em>
          </div>
          <button
            className="rounded-md w-[200px] text-center"
            onClick={(_) => setIsUpdateClaimStatusOpened(true)}
            data-testid="update-claim-btn"
          >
            <div className="space-x-2 flex justify-center items-center bg-mPrimary px-4 py-2 rounded text-white">
              <span className="text-base">Update Claim Status</span>
            </div>
          </button>
        </header>
        <main className="flex flex-col w-full mt-6 lg:flex-row lg:justify-between lg:items-center">
          <div className="w-full mb-8 lg:w-[500px]">
            <h2 className="font-medium text-2xl mb-4">mPrimary Information</h2>
            <div className="gap-y-4 w-full grid grid-cols-2">
              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                Policy Number:
              </em>
              <em className="not-italic">123jkf5402</em>

              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                User Email:
              </em>
              <em className="not-italic break-words">
                fortunateanozie@gmail.com
              </em>

              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                Type of Claim:
              </em>
              <em className="not-italic">Auto</em>

              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                Bill Estimate:
              </em>
              <em className="not-italic">{nairaSign}50,000.00</em>

              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                Date:
              </em>
              <em className="not-italic">May 7, 2023</em>

              <em className="not-italic text-[##4F4F4F] text-sm inline-block mr-4">
                Description:
              </em>
              <p>
                While I was driving my car got hit by some mad keke-napep. He
                rear-ended me and damaged my bumper.
              </p>
            </div>
          </div>
          <div className="border rounded-md border-[#ccc] p-8 w-full lg:w-[400px] xl:w-[500px] mx-auto lg:mx-0">
            <h2 className="font-medium text-2xl text-[#333] mb-6">Images</h2>
            <Slider className="w-full" {...settings}>
              <div className="mx-auto">
                <img className="mx-auto lg:w-full" src={car} />
              </div>

              <div className="mx-auto">
                <img className="mx-auto lg:w-full" src={car} />
              </div>

              <div className="mx-auto">
                <img className="mx-auto lg:w-full" src={car} />
              </div>

              <div className="mx-auto">
                <img className="mx-auto lg:w-full" src={car} />
              </div>
            </Slider>
          </div>
        </main>
      </div>
    </>
  );
};
