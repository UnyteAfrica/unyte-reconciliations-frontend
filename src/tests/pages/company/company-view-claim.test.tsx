import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { ViewClaimsPage } from "@/pages/company/view-claims";
import { OverlayContextProvider } from "@/context/overlay.context";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

function setup(reactNode: React.ReactNode) {
  const queryClient = new QueryClient();
  return {
    user: userEvent.setup(),
    ...render(
      <QueryClientProvider client={queryClient}>
        <OverlayContextProvider>{reactNode}</OverlayContextProvider>
      </QueryClientProvider>
    ),
  };
}

describe("View Claims", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <ViewClaimsPage />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should show update claim overlay when update claim status button is clicked", async () => {
    const { user } = setup(
      <BrowserRouter>
        <ViewClaimsPage />
      </BrowserRouter>
    );

    const updateBtn = screen.getByText("Update Claim Status");
    await user.click(updateBtn);
    const processingTxt = screen.getByText("Processing");
    const completedTxt = screen.getByText("Completed");
    expect(processingTxt).toBeVisible();
    expect(completedTxt).toBeVisible();
  });

  it("should close update claim overlay when click is made outside of the box", async () => {
    const { user } = setup(
      <BrowserRouter>
        <ViewClaimsPage />
      </BrowserRouter>
    );

    const updateBtn = screen.getByText("Update Claim Status");
    await user.click(updateBtn);
    const processingTxt = screen.getByText("Processing");
    const completedTxt = screen.getByText("Completed");
    expect(processingTxt).toBeVisible();
    expect(completedTxt).toBeVisible();

    const overlay = screen.getByTestId("overlay");
    await user.click(overlay);
    const hiddenProcessing = screen.queryByText("Processing");
    const hiddenCompleted = screen.queryByText("Completed");
    expect(hiddenProcessing).toBeNull();
    expect(hiddenCompleted).toBeNull();
  });
});
