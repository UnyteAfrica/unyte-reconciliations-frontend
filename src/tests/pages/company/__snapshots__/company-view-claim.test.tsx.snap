// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`View Claims > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-8 max-w-6xl mx-auto w-full"
      >
        <header
          class="flex pb-4 border-b border-[#e0e0e0] flex-col lg:flex-row lg:w-full lg:justify-between"
        >
          <div
            class="flex flex-col space-y-3 mb-4"
          >
            <div
              class="flex items-center"
            >
              <em
                class="font-medium text-xl inline-block mr-4 not-italic"
              >
                123jkf5402
              </em>
              <div
                class="p-2 py-1 font-medium text-sm w-20 text-center bg-[#4f4f4f]/[.1]"
              >
                Submitted
              </div>
            </div>
            <em
              class="not-italic font-medium text-sm"
            >
              May 7, 2023
            </em>
          </div>
          <button
            class="rounded-md w-[200px] text-center"
            data-testid="update-claim-btn"
          >
            <div
              class="space-x-2 flex justify-center items-center bg-mPrimary px-4 py-2 rounded text-white"
            >
              <span
                class="text-base"
              >
                Update Claim Status
              </span>
            </div>
          </button>
        </header>
        <main
          class="flex flex-col w-full mt-6 lg:flex-row lg:justify-between lg:items-center"
        >
          <div
            class="w-full mb-8 lg:w-[500px]"
          >
            <h2
              class="font-medium text-2xl mb-4"
            >
              mPrimary Information
            </h2>
            <div
              class="gap-y-4 w-full grid grid-cols-2"
            >
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                Policy Number:
              </em>
              <em
                class="not-italic"
              >
                123jkf5402
              </em>
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                User Email:
              </em>
              <em
                class="not-italic break-words"
              >
                fortunateanozie@gmail.com
              </em>
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                Type of Claim:
              </em>
              <em
                class="not-italic"
              >
                Auto
              </em>
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                Bill Estimate:
              </em>
              <em
                class="not-italic"
              >
                â‚¦
                50,000.00
              </em>
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                Date:
              </em>
              <em
                class="not-italic"
              >
                May 7, 2023
              </em>
              <em
                class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
              >
                Description:
              </em>
              <p>
                While I was driving my car got hit by some mad keke-napep. He rear-ended me and damaged my bumper.
              </p>
            </div>
          </div>
          <div
            class="border rounded-md border-[#ccc] p-8 w-full lg:w-[400px] xl:w-[500px] mx-auto lg:mx-0"
          >
            <h2
              class="font-medium text-2xl text-[#333] mb-6"
            >
              Images
            </h2>
            <div
              class="slick-slider w-full slick-initialized"
              dir="ltr"
            >
              <button
                class="slick-arrow slick-prev"
                data-role="none"
                style="display: block;"
                type="button"
              >
                 
                Previous
              </button>
              <div
                class="slick-list"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="-1"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="false"
                    class="slick-slide slick-active slick-current"
                    data-index="0"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="1"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="2"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="3"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="4"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="5"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="6"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="7"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        class="mx-auto"
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <img
                          class="mx-auto lg:w-full"
                          src="/src/assets/images/car.png"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="slick-arrow slick-next"
                data-role="none"
                style="display: block;"
                type="button"
              >
                 
                Next
              </button>
              <ul
                class="slick-dots"
                style="display: block;"
              >
                <li
                  class="slick-active"
                >
                  <button>
                    1
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    2
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    3
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    4
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-8 max-w-6xl mx-auto w-full"
    >
      <header
        class="flex pb-4 border-b border-[#e0e0e0] flex-col lg:flex-row lg:w-full lg:justify-between"
      >
        <div
          class="flex flex-col space-y-3 mb-4"
        >
          <div
            class="flex items-center"
          >
            <em
              class="font-medium text-xl inline-block mr-4 not-italic"
            >
              123jkf5402
            </em>
            <div
              class="p-2 py-1 font-medium text-sm w-20 text-center bg-[#4f4f4f]/[.1]"
            >
              Submitted
            </div>
          </div>
          <em
            class="not-italic font-medium text-sm"
          >
            May 7, 2023
          </em>
        </div>
        <button
          class="rounded-md w-[200px] text-center"
          data-testid="update-claim-btn"
        >
          <div
            class="space-x-2 flex justify-center items-center bg-mPrimary px-4 py-2 rounded text-white"
          >
            <span
              class="text-base"
            >
              Update Claim Status
            </span>
          </div>
        </button>
      </header>
      <main
        class="flex flex-col w-full mt-6 lg:flex-row lg:justify-between lg:items-center"
      >
        <div
          class="w-full mb-8 lg:w-[500px]"
        >
          <h2
            class="font-medium text-2xl mb-4"
          >
            mPrimary Information
          </h2>
          <div
            class="gap-y-4 w-full grid grid-cols-2"
          >
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              Policy Number:
            </em>
            <em
              class="not-italic"
            >
              123jkf5402
            </em>
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              User Email:
            </em>
            <em
              class="not-italic break-words"
            >
              fortunateanozie@gmail.com
            </em>
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              Type of Claim:
            </em>
            <em
              class="not-italic"
            >
              Auto
            </em>
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              Bill Estimate:
            </em>
            <em
              class="not-italic"
            >
              â‚¦
              50,000.00
            </em>
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              Date:
            </em>
            <em
              class="not-italic"
            >
              May 7, 2023
            </em>
            <em
              class="not-italic text-[##4F4F4F] text-sm inline-block mr-4"
            >
              Description:
            </em>
            <p>
              While I was driving my car got hit by some mad keke-napep. He rear-ended me and damaged my bumper.
            </p>
          </div>
        </div>
        <div
          class="border rounded-md border-[#ccc] p-8 w-full lg:w-[400px] xl:w-[500px] mx-auto lg:mx-0"
        >
          <h2
            class="font-medium text-2xl text-[#333] mb-6"
          >
            Images
          </h2>
          <div
            class="slick-slider w-full slick-initialized"
            dir="ltr"
          >
            <button
              class="slick-arrow slick-prev"
              data-role="none"
              style="display: block;"
              type="button"
            >
               
              Previous
            </button>
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-1"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="1"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="2"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="3"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="4"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="5"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="6"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="7"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="mx-auto"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <img
                        class="mx-auto lg:w-full"
                        src="/src/assets/images/car.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="slick-arrow slick-next"
              data-role="none"
              style="display: block;"
              type="button"
            >
               
              Next
            </button>
            <ul
              class="slick-dots"
              style="display: block;"
            >
              <li
                class="slick-active"
              >
                <button>
                  1
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  2
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  3
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  4
                </button>
              </li>
            </ul>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
